rpcclient/discover.go

```diff
@@ -65,7 +65,7 @@ func (c *Client) QueryAll(provides types.ProvidesCoin, searchTime uint64) ([]*rp
 		return nil, resp.Error
 	}
 
-	var res *rpctypes.QueryAllResponse
+	var res rpctypes.QueryAllResponse
 	if err = json.Unmarshal(resp.Result, &res); err != nil {
```

Comment:
I didn't understand how this code was working.  I thought the object passed to `Unmarshal` had to already be allocated, but maybe it didn't need to be fixed. If you pass a pointer to a pointer, will `json.Unmarshal` allocate the object?

Reply:
it should work either way, it worked before so I think both are ok